tests/app/test_ipexing.py

```diff
@@ -919,3 +925,314 @@ def test_granter(helpers):
+                                   'p': '',
+                                   'q': {},
+                                   'r': '/ipex/apply',
+                                   't': 'exn',
+                                   'v': 'KERI10JSON000187_'}
+        assert end == b''
+        sigs = ["AAAa70b4QnTOtGOsMqcezMtVzCFuRJHGeIMkWYHZ5ZxGIXM0XDVAzkYdCeadfPfzlKC6dkfiwuJ0IzLOElaanUgH"]
+
+        body = dict(
+            exn=applySerder.ked,
+            sigs=sigs,
+            atc="",
+            rec=["EZ-i0d8JZAoTNZH3ULaU6JR2nmwyvYAfSVPzhzS6b5CM"]
+        )
+
+        data = json.dumps(body).encode("utf-8")
+        res = client.simulate_post(path="/identifiers/test/ipex/apply", body=data)
+        
+        assert res.status_code == 400
+        assert res.json == {'description': 'attempt to send to unknown '
+                                           'AID=EZ-i0d8JZAoTNZH3ULaU6JR2nmwyvYAfSVPzhzS6b5CM',
+                            'title': '400 Bad Request'}
+
+        body = dict(
+            exn=applySerder.ked,
+            sigs=sigs,
+            atc="",
+            rec=[pre1]
+        )
+
+        # Bad Sender
+        data = json.dumps(body).encode("utf-8")
+        res = client.simulate_post(path="/identifiers/BAD/ipex/apply", body=data)
+        assert res.status_code == 404
+        assert res.json == {'description': 'alias=BAD is not a valid reference to an identifier',
+                            'title': '404 Not Found'}
+
+        data = json.dumps(body).encode("utf-8")
+        res = client.simulate_post(path="/identifiers/test/ipex/apply", body=data)
+        assert res.json == {'done': False,
+                            'error': None,
+                            'metadata': {'said': 'EJq6zSDUWw6iaBz8n1cY5cAW3Rrgp4E3sUsoz5JkoMZc'},
+                            'name': 'exchange.EHgwVwQT15OJvilVvW57HE4w0-GPs_Stj2OFoAHZSysY',
+                            'response': None}
+
+        assert res.status_code == 200
+        assert len(agent.exchanges) == 1
+
+
+def test_ipex_offer(helpers, mockHelpingNowIso8601):
+    with helpers.openKeria() as (_, agent, app, client):
+        offerEnd = ipexing.IpexOfferCollectionEnd()
+        app.add_route("/identifiers/{name}/ipex/offer", offerEnd)
+
+        end0 = aiding.IdentifierCollectionEnd()
+        app.add_route("/identifiers", end0)
+        aidEnd = aiding.IdentifierResourceEnd()
+        app.add_route("/identifiers/{name}", aidEnd)
+
+        salt = b'0123456789abcdef'
+        op = helpers.createAid(client, "test", salt)
+        aid = op["response"]
+        pre = aid['i']
+        assert pre == "EHgwVwQT15OJvilVvW57HE4w0-GPs_Stj2OFoAHZSysY"
+
+        salt2 = b'0123456789abcdeg'
+        op = helpers.createAid(client, "recp", salt2)
+        aid1 = op["response"]
+        pre1 = aid1['i']
+        assert pre1 == "EFnYGvF_ENKJ_4PGsWsvfd_R6m5cN-3KYsz_0mAuNpCm"
+
+        # This should be a metadata ACDC in reality
```

Comment:
Question here... if a metadata ACDC exists for making commitments to the metadata of a private ACDC - does this mean for a public ACDC you offer the full public ACDC or still the metadata ACDC?

Reply:
I think it has to be the full public ACDC